File: package/main/src/Date/subBusinessDays.ts
import { addBusinessDays } from "./addBusinessDays";

/**
 * Subtracts business days from a date.
 * Equivalent to addBusinessDays(date, -amount, holidays).
 *
 * @param {Date} date - Base date
 * @param {number} amount - Business days to subtract
 * @param {Date[]} [holidays=[]] - Optional holiday dates
 * @returns {Date} A new Date after walking backward `amount` business days
 * @example
 * subBusinessDays(new Date(2025, 3, 21), 1); // 2025-04-18 (Friday)
 */
export const subBusinessDays = (
  date: Date,
  amount: number,
  holidays: Date[] = [],
): Date => addBusinessDays(date, -amount, holidays);

File: package/umt_python/src/date/sub_business_days.py
from datetime import datetime

from .add_business_days import add_business_days


def sub_business_days(
    date: datetime,
    amount: int | float,
    holidays: list[datetime] | None = None,
) -> datetime:
    """
    Subtract business days from a date.

    Equivalent to add_business_days(date, -amount, holidays).

    Args:
        date: Base date
        amount: Business days to subtract
        holidays: Optional holiday dates

    Returns:
        A new datetime after walking backward `amount` business days

    Example:
        >>> from datetime import datetime
        >>> sub_business_days(datetime(2025, 4, 21, 9, 30), 1)
        datetime.datetime(2025, 4, 18, 9, 30)
    """
    return add_business_days(date, -amount, holidays)
